### PR TITLE
Journal based on events count and size

### PR DESCRIPTION
**Pull request by xbrianh**
This attempts to journal new events based on minimum number of events per journal, and the minimum size of the combined event data. Unless both conditions are met, raise an exception.

### CODE REVIEW

File: flashflood/__init__.py [hunk 1]
```diff
@@ -69,15 +69,30 @@ def put(self, data, event_id: str=None, date: datetime=None):
         self._upload_journal(_Journal(journal_id, manifest, io.BytesIO(data)), is_new=True)
         return Event(event_id, date, data)
 
-    def journal(self, number_of_events=10):
+    def journal(self, minimum_number_of_events=100, minimum_size=None):
```
Comment (DailyDreaming) [inline flashflood/__init__.py]: Readme still says `ff.journal(number_of_events=5000)`.

Also, I'd consider adding typing, i.e. (` def journal(self, minimum_number_of_events: int =100, minimum_size: int = None)`), both here and elsewhere.

File: flashflood/__init__.py [hunk 2]
```diff
@@ -69,15 +69,30 @@ def put(self, data, event_id: str=None, date: datetime=None):
...
+        assert 2 <= minimum_number_of_events
```
Comment (DailyDreaming) [inline flashflood/__init__.py]: Why was the hard minimum set to `3`?  (I maybe would expect `1`?)

Comment (xbrianh) [reply flashflood/__init__.py]: Using `1` would result in a null-op

File: flashflood/__init__.py (hunk 1 above)
Comment (xbrianh) [reply flashflood/__init__.py]: `README.md` updated.

I'll consider adding type hints in a future PR.

**Comment by xbrianh** [comment]
@DailyDreaming `journal()` doesn't "break" if the `minimum_number_of_events` and `minimum_size` events are not met, it throws an exception which can be caught. An alternative design is using a return value to indicate whether or not a journal was created. In either case, the idea is to call `journal` repeatedly until all journals have been created.

**Pull request merged**